controller/sync.go

```diff
@@ -323,6 +332,15 @@ func (m *appStateManager) SyncAppState(app *v1alpha1.Application, state *v1alpha
 	state.Phase, state.Message, resState = syncCtx.GetState()
 	state.SyncResult.Resources = nil
 	for _, res := range resState {
+
+		err := argo.FormatSyncMsg(&res, func(kind string) (*[]kube.APIResourceInfo, error) {
+			return m.getAPIResourceVersion(app.Spec.Destination.Server, kind)
```

Comment:
Can we cache this using a var declared outside the loop? No need to do the same work repeatedly.

Reply:
I need to check the sync status message for every resource, therefore `FormatSyncMsg ` will be loop over `resState`. But In the Util function, I'm caching the [APIResource](https://github.com/argoproj/argo-cd/pull/12980/files#diff-87b3b573e8c40af95af4dbda83f7dbc467219b02182415ff8e1f0ad73a82dc59R37) response.  
I add [logic](https://github.com/argoproj/argo-cd/pull/12980/files#diff-87b3b573e8c40af95af4dbda83f7dbc467219b02182415ff8e1f0ad73a82dc59R47) to refresh every 5 mins(time elapsed since last refresh is great then 5 min)

Reply:
Agreed